contracts/src/v0.1/L2endpoint.sol

```diff
@@ -53,4 +72,109 @@ contract L2Endpoint is Ownable {
         IL2Aggregator(_aggregator).submit(_roundId, _submission);
         emit Submitted(_roundId, _submission);
     }
+
+    function computeRequestId(
+        bytes32 keyHash,
+        address sender,
+        uint64 accId,
+        uint64 nonce
+    ) private pure returns (uint256, uint256) {
+        uint256 preSeed = uint256(keccak256(abi.encode(keyHash, sender, accId, nonce)));
+        uint256 requestId = uint256(keccak256(abi.encode(keyHash, preSeed)));
+        return (requestId, preSeed);
+    }
+
+    function requestRandomWords(
+        bytes32 keyHash,
+        uint64 accId,
+        uint32 callbackGasLimit,
+        uint32 numWords,
+        bool isDirectPayment
+    ) private returns (uint256) {
```

Comment:
It would be great if we can validate whether the account at least exists. Explained in one other comment.